_posts/2019-08-26-how-to-use-gruntwork-infrastructure-as-code-library.adoc

```diff
@@ -1399,6 +1409,272 @@ module "vpc" {
+}
+
+module "sns" {
+  source = "git::git@github.com:gruntwork-io/package-messaging.git//modules/sns?ref=v0.3.2"
+  name         = "my-topic"
+  display_name = "MyTopic"
+}
+----
+
+The `git::git@github.com:gruntwork-io` portion of the `source` attribute indicates that this module is accessed over SSH.
+Therefore, to access the Gruntwork module, the system running Terraform (the TFC executor in this case) will need permissions to clone the repository
+via SSH.
+
+Fortunately, creating a workspace that uses a Gruntwork module is easily accomplished in TFC. Take the following steps:
+
+1. First, if you don't have one already, you'll need a machine user that has access to Gruntwork. A machine user is an
+account that is only used for automation, and is not used by humans. In this case, the "machine" in question is the TFC
+executor. Create a new Github user, and email the username to mailto:support@gruntwork.io[support@gruntwork.io]. We'll make sure the user has access to our repositories.
+
+1. Next, add an SSH key to the new GitHub machine user. GitHub has https://help.github.com/en/enterprise/2.19/user/github/authenticating-to-github/connecting-to-github-with-ssh[easy-to-follow instructions].
+
+1. Now, add the private portion of the SSH key to TFC. You'll find the option under SSH Keys in the TFC organization settings. We called ours _Gruntwork access_.
+
+1. Set up a new workspace and link it to your code repository that uses Gruntwork's module as a source.
+
+1. In the Workspace settings, in the SSH key section, you can now choose the Gruntwork access key.
+
+.Configuring a TFC Workspace to access Gruntwork repositories
+image::/assets/img/guides/infrastructure-as-code-library/tfc_ssh_key.png[]
+
+
+When TFC runs your module, it will use the configured SSH key to clone the Gruntwork repository during the run, and will continue with creating the SNS topic you defined. You can reuse this SSH key across any other workspaces that depend on Gruntwork modules.
+
+That's it! You can use this pattern for any of Gruntwork's Terraform modules. This even works for modules like github.com/gruntwork-io/terraform-aws-eks[`terraform-aws-eks`] which need to download external dependencies (like https://github.com/gruntwork-io/kubergrunt[`kubergrunt`]). Those modules use the special https://github.com/gruntwork-io/package-terraform-utilities/tree/master/modules/executable-dependency[`executable-dependency`] module to install the external dependency within the executor at run time.
+
+In the future, Gruntwork may offer modules through  a private Terraform registry to further simplify the process.
+
+[[terragrunt_tfc_tfe]]
+==== Using Terragrunt with TFC and TFE
+Configuring Terragrunt to work with TFC/TFE is a bit more involved. As mentioned previously, TFC and TFE execute
+`terraform` commands under the hood, and the executors that run these commands aren't able to execute the Terragrunt
+binary. Running Terragrunt directly through the TFC UI isn't an option.
```

Comment:
I'd rephrase this slightly:

1. The TFC/TFE UI can only run the Terraform binary, whereas Terragrunt expects you to run it, and it, in turn, will run Terraform. 
1. Therefore, there's no way currently to use the TFC/TFE UI to trigger deployments.
1. However, you can use TFC/TFE to [run remote operations](https://www.terraform.io/docs/backends/operations.html) such as `plan` and `apply`. That is, you use Terragrunt to organize your code and keep it DRY, and you can configure it so that when you run `terragrunt apply` on your computer or in a CI server, it runs `terraform apply` in TFC/TFE rather than your own computer. Under the hood, you can create the workspace automatically, or, re-use a workspace that already exists, including variables and env vars already set in it, such as your AWS access keys, so you don't need to have those locally. You'll also be able to see the `apply` output in the web UI (but not trigger it from the web UI) and see the history of all `apply` calls in the web UI.